app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :email, format: URI::MailTo::EMAIL_REGEXP
+
+  # the authenticate method from devise documentation
+  def self.authenticate(email, password)
+    password_param = password
+
+    user = User.find_for_authentication(email: email)
+    user&.valid_password?(password_param) ? user : nil
+  end
+end
```

Comment:
Could you point me out why do we need to add this? 🤔

Reply:
So it's something that Devise needs to find a user when we have their credentials 😄, you can see the documentation here: https://github.com/heartcombo/devise/wiki/How-To:-Find-a-user-when-you-have-their-credentials

Line 13 is mostly for the CI 😂